src/main/java/com/tabnine/binary/requests/autocomplete/CompletionPostprocess.kt

```diff
@@ -0,0 +1,48 @@
+package com.tabnine.binary.requests.autocomplete
+
+fun postprocess(request: AutocompleteRequest, result: AutocompleteResponse, tabSize: Int) {
+    result.results.forEach { it.new_prefix = it.new_prefix.replace("\t", " ".repeat(tabSize)) }
+
+    val requestIndentation = lastLineIndentation(request.before.replace("\t", " ".repeat(tabSize))) ?: return
+    if (requestIndentation == 0) return
```

Comment:
If you have something like that prefix:
```kotlin
/**
 * Finds the amount of spaces or tabs in the last line of the given `value`,
 * returning `null` if `value` has no newlines, or the last line is not whitespaces only.
 */
fun lastLineIndentation(value: String): Int? {
    try {
        val lastLine = value.lines().last()
```

Where would the following snippet end?
```kotlin
        if (lastLine.isBlank()) {
            return lastLine.length
        }
    } catch (e: java.util.NoSuchElementException) {
        // do nothing
    }
    return null
}

/**
 * Constructs a regex which accepts a \n followed by at most `indentation - 1` spaces,
 * followed by any text or another \n.
 */
fun constructRegex(indentation: Int): Regex {
    val upperLimit = indentation - 1
    return Regex("""^ {0,$upperLimit}[\w\n]+""", RegexOption.MULTILINE)
}
```

Reply:
its not clear where exactly is the cursor in your example, but assuming you're here:
```kotlin
/**
 * Finds the amount of spaces or tabs in the last line of the given `value`,
 * returning `null` if `value` has no newlines, or the last line is not whitespaces only.
 */
fun lastLineIndentation(value: String): Int? {
    try {
        val lastLine = value.lines().last()
        //<-- cursor here
```

then the snippet will end after the closing bracket of he `if` statement - before the `catch` statement.